prow/tide/tide.go

```diff
@@ -995,12 +995,18 @@ func (c *Controller) trigger(sp subpool, presubmits map[int][]config.Presubmit,
 		BaseSHA: sp.sha,
 	}
 	for _, pr := range prs {
+		changes := c.changedFiles.prChanges(&pr)
+		changedFiles, err := changes()
```

Comment:
This comment https://github.com/kubernetes/test-infra/pull/12751#issuecomment-499263182 suggested that Tide doesn't use the GitHub client to fetch PR changes, but that is what is happening here.

If we are triggering a batch of PRs (the expensive case for this) we will have just checked out all the PRs to make sure they don't conflict so not only will using the git client save API tokens in this case, it will also be very fast. Please do that instead.

Reply:
Thank you for pointing this out. Looking at the logic again, since we are going to use the git client to get the list of changed files here, do you think it is a good idea to just make the `changedFiles` struct use the git client and avoid the `GetPullRequestChanges` call altogether in `tide`? (If we can avoid making multiple `Clone` calls for the same PR, token usage may also improve).

Reply:
Yeah that makes sense to me.

Reply:
Thank you. I have created a separate PR for that: #13731